dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -197,6 +236,81 @@ protected void generateSnappyConfig() {
         }
     }
 
+    protected void generateSnappyLocatorConfig() {
+        if (logDirExists) return;
+        else {
+            String addr = HostHelper.getHostAddress();
+            int port = PortHelper.getRandomPort();
+            String endpoint = addr + ":" + port;
+            String clientPort = " -client-port=";
+            String dirPath = snappyTest.getLogDir();
+            String locatorLogDir = HostHelper.getLocalHost() + " -dir=" + dirPath + clientPort + port;
+            if (locatorLogDir == null) {
+                String s = "Unable to find " + RemoteTestModule.getMyClientName() + " log directory path for writing to the locators file under conf directory";
+                throw new TestException(s);
+            }
+            SnappyBB.getBB().getSharedMap().put("locatorLogDir" + "_" + snappyTest.getMyTid(), locatorLogDir);
+            String portString = port + "";
+            SnappyBB.getBB().getSharedMap().put("locatorHost", HostHelper.getLocalHost());
+            SnappyBB.getBB().getSharedMap().put("locatorPort", portString);
```

Comment:
simply use
SnappyBB.getBB().getSharedMap().put("locatorPort", port.toString());

No need to create extra variable portString